File: 12908/book.py
import sys
import math

# Time limit exceeded

def get_nearest_n_above(S):
	# y = n + 0.5
	y = math.sqrt(2*S + 0.25)
	return math.floor(y - 0.25) + 1

def get_sum_till_N(n):
	return (n * (n+1)) // 2

while True:
	n = int(sys.stdin.readline())

	if n == 0:
		break

	nearest_n = get_nearest_n_above(n)
	sum_till_nearest_n = get_sum_till_N(nearest_n)

	# print("nearest:", nearest_n)
	# print("sum nearest:", sum_till_nearest_n)

	missing_num = sum_till_nearest_n - n

	print(missing_num, nearest_n)

File: 12908/book.c
#include <stdio.h>
#include <math.h>

// Accepted

long get_nearest_n(long S)
{
	double y = sqrt(2 * S + 0.25);
	return floor(y - 0.5) + 1;
}

long get_sum_till_n(long n)
{
	return (n * (n + 1)) / 2;
}

int main(int argc, char *argv[])
{
	long s, n, s_n, missing_num;

	while(1)
	{
		scanf("%ld", &s);

		if (s == 0)
			break;

		n = get_nearest_n(s);
		s_n = get_sum_till_n(n);
		missing_num = s_n - s;

		printf("%ld %ld\n", missing_num, n);
	}

	return 0;
}